packages/nimble-components/src/select/index.ts

```diff
@@ -49,51 +146,701 @@ export class Select extends FoundationSelect implements ErrorPattern {
+        const currentBox = this.getBoundingClientRect();
+        const viewportHeight = window.innerHeight;
+        const availableBottom = viewportHeight - currentBox.bottom;
+
+        if (this.forcedPosition) {
+            this.position = this.positionAttribute;
+        } else if (currentBox.top > availableBottom) {
+            this.position = SelectPosition.above;
+        } else {
+            this.position = SelectPosition.below;
+        }
+
+        this.positionAttribute = this.forcedPosition
+            ? this.positionAttribute
+            : this.position;
+
+        this.maxHeight = this.position === SelectPosition.above
+            ? Math.trunc(currentBox.top)
+            : Math.trunc(availableBottom);
         this.updateListboxMaxHeightCssVariable();
     }
 
-    // Workaround for https://github.com/microsoft/fast/issues/5773
     public override slottedOptionsChanged(
         prev: Element[],
         next: Element[]
     ): void {
         const value = this.value;
+        this.options.forEach(o => {
+            const notifier = Observable.getNotifier(o);
+            notifier.unsubscribe(this, 'value');
+        });
+
         super.slottedOptionsChanged(prev, next);
+
+        this.options.forEach(o => {
+            const notifier = Observable.getNotifier(o);
+            notifier.subscribe(this, 'value');
+        });
+        this.setProxyOptions();
+        this.updateValue();
+        // We need to force an update to the filteredOptions observable
+        // (by calling 'filterOptions()) so that the template correctly updates.
+        this.filterOptions();
         if (value) {
             this.value = value;
         }
+        this.committedSelectedOption = this.options[this.selectedIndex];
     }
 
-    private regionChanged(
-        _prev: AnchoredRegion | undefined,
-        _next: AnchoredRegion | undefined
+    public override clickHandler(e: MouseEvent): BooleanOrVoid {
+        // do nothing if the select is disabled
+        if (this.disabled) {
+            return;
+        }
+
+        if (this.open) {
+            const captured = (e.target as HTMLElement).closest<ListOption>(
+                'option,[role=option]'
+            );
+
+            if (!captured?.disabled) {
+                this.updateSelectedIndexFromFilteredSet();
+            }
+
+            if (captured?.disabled) {
+                return;
+            }
+        }
+
+        super.clickHandler(e);
+
+        this.open = this.collapsible && !this.open;
+
+        this.focusFilterInput();
+
+        if (!this.open && this.indexWhenOpened !== this.selectedIndex) {
+            this.updateValue(true);
+        }
+    }
+
+    /**
+     * Updates the value when an option's value changes.
+     *
+     * @param source - the source object
+     * @param propertyName - the property to evaluate
+     *
+     * @internal
+     * @override
+     */
+    public override handleChange(source: unknown, propertyName: string): void {
```

Comment:
@jattasNI @rajsite, this and the following 2 overrides are copied directly from FAST.